core/api/src/main/java/org/trellisldp/api/MutableDataService.java

```diff
@@ -27,27 +28,39 @@
 public interface MutableDataService<U> extends RetrievalService<U> {
 
     /**
-     * @param identifier the identifier for the resource to persist
+     * Put a resource into the server.
+     *
+     * @param identifier the identifier for the new resource
      * @param session the session context for this operation
-     * @param resource a resource to persist
-     * @return whether the resource was successfully persisted
+     * @param ixnModel the LDP interaction model for this resource
+     * @param container an LDP container for this resource
```

Comment:
One more thing -- it would be really helpful if the javadoc here indicates that the container parameter can be `null`.

Reply:
Good catch. What about `javax.annotation.CheckForNull` or `javax.annotation.Nullable`? Worth adding?

Reply:
I haven't been adding those annotations to other parts of the code, but it seems like a good idea for the API. Feel free to start here, and the relevant annotations can be added elsewhere by the by.

Reply:
Hm, never mind, I have to walk it back. Those annotations aren't in our classpath, or at least not in such a way as to let me deploy them here. Oh, well, Javadocs are good too.

Reply:
Just a note that these annotations `javax.annotation.Nullable` are not part of the standard `annotation-api` artifact. We'd need to add a dependency on 'com.google.code.findbugs/jsr305' (JSR 305 itself seems pretty dormant). Is it worth adding that dependency?

Reply:
Right, see my note above (we just crossed messages). If it _is_ worth adding that, it seems to me not to belong in this PR, eh?

Reply:
correct -- that would be a separate PR.